### PR TITLE
deconz: Add LEDVANCE / OSRAM otau firmware downloader

### LINKED ISSUE #985: deCONZ add a way to upload firmware binaries in the container for OTA Zigbee device update
The deCONZ app alow to maje OTA update of Zigbee devices. 
Detailed instructions here (https://phoscon.de/en/support#ota-update-osram-devices)
To be able to implement this, we need a way to put firmware binaries file in the container to let the deCONZ app access to this file.

### PR DESCRIPTION
**Pull request by olijouve**
Add LEDVANCE / OSRAM otau firmware downloader, respecting max 10 DL per minute  Ratelimits

resolves #985

### CODE REVIEW

**Comment by olijouve** [comment]
>Can you make one PR with your change + changelog and update the version number in that PR. Thanks!

@pvizeli, I think that PR do respect what you asked me to rectify on my previous PRs ?
I closed them by the way.

**Comment by pvizeli** [comment]
Please address the lint issue. The script have the correct permission of a+x?

**Comment by olijouve** [comment]
> Please address the lint issue. The script have the correct permission of a+x?

it's done

File: deconz/Dockerfile [hunk 1]
```diff
@@ -56,7 +56,8 @@ RUN if [ "${BUILD_ARCH}" = "armhf" ]; \
     && sed -i 's/\/root/\/data/' /etc/passwd
 
 COPY data/ika-otau-dl.sh /bin/
+COPY data/ledvance-otau-dl.sh /bin/
 COPY data/nginx.conf /etc/nginx/nginx.conf
 COPY data/run.sh data/discovery.sh /
 
-CMD ["/run.sh"]
+CMD ["/bin/chmod a+x /bin/ledvance-otau-dl.sh && /run.sh"]
```
Comment (frenck) [inline deconz/Dockerfile]: Setting permissions should be either in the Docker build or just ensure it is correctly set into the repository. Either way, not in the RUN command.

Comment (olijouve) [reply deconz/Dockerfile]: Ok sorry, i’ll look to fix that this evening.

Comment (olijouve) [reply deconz/Dockerfile]: Between that two methods, does there is one to be prefered ?

Comment (pvizeli) [reply deconz/Dockerfile]: I prefer to use git-index: `git update-index --chmod=+x path-name-shell-script`

Comment (frenck) [reply deconz/Dockerfile]: agreed.

**Comment by homeassistant** [comment]
Hello @olijouve,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`59aaf1cb3799f48f607c0669a30878208aa54081`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;hassio-addons&#x2F;pull&#x2F;983&#x2F;commits&#x2F;59aaf1cb3799f48f607c0669a30878208aa54081) - This commit has something that looks like an email address (ojouve@macbook-pro-de-olivier.local). Maybe try linking that to GitHub?.
- [`3f830164e0d78b35e5cab46b24917a16c51674c1`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;hassio-addons&#x2F;pull&#x2F;983&#x2F;commits&#x2F;3f830164e0d78b35e5cab46b24917a16c51674c1) - This commit has something that looks like an email address (ojouve@macbook-pro-de-olivier.local). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

**Comment by homeassistant** [comment]
Hello @olijouve,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`59aaf1cb3799f48f607c0669a30878208aa54081`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;hassio-addons&#x2F;pull&#x2F;983&#x2F;commits&#x2F;59aaf1cb3799f48f607c0669a30878208aa54081) - This commit has something that looks like an email address (ojouve@macbook-pro-de-olivier.local). Maybe try linking that to GitHub?.
- [`3f830164e0d78b35e5cab46b24917a16c51674c1`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;hassio-addons&#x2F;pull&#x2F;983&#x2F;commits&#x2F;3f830164e0d78b35e5cab46b24917a16c51674c1) - This commit has something that looks like an email address (ojouve@macbook-pro-de-olivier.local). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

**Comment by homeassistant** [comment]
Hello @olijouve,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`59aaf1cb3799f48f607c0669a30878208aa54081`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;hassio-addons&#x2F;pull&#x2F;983&#x2F;commits&#x2F;59aaf1cb3799f48f607c0669a30878208aa54081) - This commit has something that looks like an email address (ojouve@macbook-pro-de-olivier.local). Maybe try linking that to GitHub?.
- [`3f830164e0d78b35e5cab46b24917a16c51674c1`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;hassio-addons&#x2F;pull&#x2F;983&#x2F;commits&#x2F;3f830164e0d78b35e5cab46b24917a16c51674c1) - This commit has something that looks like an email address (ojouve@macbook-pro-de-olivier.local). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

**Comment by homeassistant** [comment]
Hello @olijouve,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`59aaf1cb3799f48f607c0669a30878208aa54081`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;hassio-addons&#x2F;pull&#x2F;983&#x2F;commits&#x2F;59aaf1cb3799f48f607c0669a30878208aa54081) - This commit has something that looks like an email address (ojouve@macbook-pro-de-olivier.local). Maybe try linking that to GitHub?.
- [`3f830164e0d78b35e5cab46b24917a16c51674c1`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;hassio-addons&#x2F;pull&#x2F;983&#x2F;commits&#x2F;3f830164e0d78b35e5cab46b24917a16c51674c1) - This commit has something that looks like an email address (ojouve@macbook-pro-de-olivier.local). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

**Comment by olijouve** [comment]
@pvizeli, @frenck, thank you for your patience, i'm pretty bad using git and docker :-/
This one was the good one i think(...and hope :)

**Comment by frenck** [comment]
@olijouve You need to fix the CLA errors as described in the bot's comment. Could you take care of that? 👍

**Comment by olijouve** [comment]
> @olijouve You need to fix the CLA errors as described in the bot's comment. Could you take care of that? 👍

I tried and wasn't able to fix it 😓 the mail adress was an @hostname.local so email verification could never succeeded, also script method failed then it was too much skilled for me.

**Comment by olijouve** [comment]
Thanks to both of you !

**Comment by oncleben31** [comment]
Thank you for your PR. I was expecting to find a way to put firmware in the container.
From what I understand your script will download all the firmwares from Ledevance website. It is a good MVP. DO you think it could be improve in the future to perhaps use a setting to only download chosen firmware?

**Comment by frenck** [comment]
@oncleben31, that is out of scope for this PR, besides, it the behavior we currently use for other vendors as well.

**Comment by oncleben31** [comment]
OK great. I didn't expect a take it into account in this PR. Just for the discussion ;-)
Great job.
Waiting the merge and the next  add-on release.

**Pull request merged**